pandas/core/algorithms.py

```diff
@@ -202,8 +202,13 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
     from pandas.tseries.period import PeriodIndex
 
     is_period = com.is_period_arraylike(values)
-    values = Series(values).values
+    values = Series(values)
     is_category = com.is_categorical_dtype(values.dtype)
+    if is_category and dropna:
+        values = values.dropna()
```

Comment:
this is better to dispatch to the Categorical (eg it's the values) and define dropna() there